*** src/greeter.c.orig	2011-07-29 15:42:33.000000000 +0200
--- src/greeter.c	2012-09-20 00:06:20.807196946 +0200
***************
*** 120,125 ****
--- 120,149 ----
  	return res;
  }
  
+ static void on_evt_request_hide_login_pane(GtkWidget *widget, gpointer user_data)
+ {
+ 	if (!gtk_widget_get_visible(prompt)
+ 		&& !gtk_widget_get_visible(login_entry)
+ 		&& !gtk_widget_get_visible(user_list))
+ 	{
+ 		GtkWidget* w;
+ 		if( (w = (GtkWidget*)gtk_builder_get_object(builder, "login_pane"))!=NULL )
+ 			gtk_widget_hide(w);
+ 	}
+ }
+ 
+ static void on_evt_request_show_login_pane(GtkWidget *widget, gpointer user_data)
+ {
+ 	if (gtk_widget_get_visible(prompt)
+ 		|| gtk_widget_get_visible(login_entry)
+ 		|| gtk_widget_get_visible(user_list))
+ 	{
+ 		GtkWidget* w;
+ 		if( (w = (GtkWidget*)gtk_builder_get_object(builder, "login_pane"))!=NULL )
+ 			gtk_widget_show(w);
+ 	}
+ }
+ 
  static void on_entry_activate(GtkEntry* entry)
  {
  	char* tmp;
***************
*** 1036,1041 ****
--- 1060,1075 ----
  
      g_signal_connect(login_entry, "activate", G_CALLBACK(on_entry_activate), NULL);
  
+ 
+ 	// Set up signals for showing and hiding login_pane
+ 	g_signal_connect(prompt,"show",G_CALLBACK(on_evt_request_show_login_pane),NULL);
+ 	g_signal_connect(login_entry,"show",G_CALLBACK(on_evt_request_show_login_pane),NULL);
+ 	g_signal_connect(user_list,"show",G_CALLBACK(on_evt_request_show_login_pane),NULL);
+ 	g_signal_connect(prompt,"hide",G_CALLBACK(on_evt_request_hide_login_pane),NULL);
+ 	g_signal_connect(login_entry,"hide",G_CALLBACK(on_evt_request_hide_login_pane),NULL);
+ 	g_signal_connect(user_list,"hide",G_CALLBACK(on_evt_request_hide_login_pane),NULL);
+ 
+ 
      sessions = (GtkWidget*)gtk_builder_get_object(builder, "sessions");
      gtk_widget_set_name(sessions, "sessions");
      fix_combobox_entry(sessions);
***************
*** 1060,1065 ****
--- 1094,1118 ----
      else
          gtk_event_box_set_visible_window(GTK_EVENT_BOX(w), FALSE);
  
+ 
+  	// Do the same for login_pane
+     if( (w = (GtkWidget*)gtk_builder_get_object(builder, "login_pane"))!=NULL )
+     {
+         /* hacks to let GtkEventBox paintable with gtk pixmap engine. */
+ #if GTK_CHECK_VERSION(2,18,0)
+         if(gtk_widget_get_app_paintable(w))
+ #else
+         if(GTK_WIDGET_APP_PAINTABLE(w))
+ #endif
+ 
+ #if GTK_CHECK_VERSION(3,0,0)
+ 		g_signal_connect(w,"draw",G_CALLBACK(on_evt_box_draw),NULL);
+ #else
+ 		g_signal_connect(w, "expose-event", G_CALLBACK(on_evt_box_expose), NULL);
+ #endif
+     }
+ 
+ 
      if( g_key_file_get_integer(config, "display", "lang", 0) == 0 )
      {
          w = (GtkWidget*)gtk_builder_get_object(builder, "lang_box");
